src/pgwire/id_alloc.rs

```diff
@@ -32,11 +32,11 @@ impl IdAllocator {
     /// Creates a new `IdAllocator` that will assign IDs between `min` and
     /// `max`, both inclusive.
     pub fn new(min: u32, max: u32) -> IdAllocator {
-        IdAllocator(Arc::new(Mutex::new(IdAllocatorInner {
+        IdAllocator(Mutex::new(IdAllocatorInner {
             next: min,
             max,
             free: VecDeque::new(),
-        })))
+        }))
```

Comment:
Not sure why this needs to be in a mutex now that it's not clone or shared? I'm probably missing where it's shared?

Reply:
Oh I guess maybe it needs to be Send in order to even get into a threadpool?

Reply:
It is still shared, in fact! The `pgwire::Server` is shared between every connection task, via the mux. But that makes the inner arc necessary, because the mux is already arc'd.